src/deprecated.jl

```diff
@@ -231,3 +231,6 @@ end
 
 @deprecate info(obj::Union{Group,File}) g_info(obj) false
 @deprecate objinfo(obj::Union{File,Object}) o_info(obj) false
+
+@deprecate_binding filename HDF5.filename
```

Comment:
I don't think this will work — it's the same binding, so it should always throw a deprecation warning.

Testing confirms:
```
$ julia-dev --project --depwarn=yes
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.6.0-DEV.1397 (2020-10-30)
 _/ |\__'_|_|_|\__'_|  |  Commit 8d55e04b69* (2 days old master)
|__/                   |

julia> using HDF5

julia> hfile = h5open("/tmp/test.h5")
🗂️ HDF5.File: read only /tmp/test.h5
└─ 📄 complexf64

julia> HDF5.filename(hfile)
WARNING: HDF5.filename is deprecated, use HDF5.filename instead.
  likely near REPL[3]:1
"/tmp/test.h5"
```

Reply:
Hmm I don't see a way to actually fix this.

Reply:
Removing the unexports from this PR.